pkg/gadgets/profile/tcprtt/tracer/tracer.go

```diff
@@ -236,22 +236,27 @@ func (t *Tracer) install() error {
 	if err != nil {
 		return fmt.Errorf("loading specs: %w", err)
 	}
+	tcpRttSpec := &tcpRTTSpecs{}
+	if err := spec.Assign(tcpRttSpec); err != nil {
+		return err
+	}
 
-	consts := map[string]interface{}{
-		"targ_ms":         t.config.useMilliseconds,
-		"targ_laddr_hist": t.config.localAddrHist,
-		"targ_raddr_hist": t.config.remoteAddrHist,
-		"targ_sport":      htons(t.config.filterLocalPort),
-		"targ_dport":      htons(t.config.filterRemotePort),
-		"targ_saddr":      t.config.filterLocalAddress,
-		"targ_daddr":      t.config.filterRemoteAddress,
-		"targ_saddr_v6":   t.config.filterLocalAddressV6,
-		"targ_daddr_v6":   t.config.filterRemoteAddressV6,
+	consts := map[*ebpf.VariableSpec]interface{}{
+		tcpRttSpec.TargMs:        t.config.useMilliseconds,
+		tcpRttSpec.TargLaddrHist: t.config.localAddrHist,
+		tcpRttSpec.TargRaddrHist: t.config.remoteAddrHist,
+		tcpRttSpec.TargSport:     htons(t.config.filterLocalPort),
+		tcpRttSpec.TargDport:     htons(t.config.filterRemotePort),
+		tcpRttSpec.TargSaddr:     t.config.filterLocalAddress,
+		tcpRttSpec.TargDaddr:     t.config.filterRemoteAddress,
+		tcpRttSpec.TargSaddrV6:   t.config.filterLocalAddressV6,
+		tcpRttSpec.TargDaddrV6:   t.config.filterRemoteAddressV6,
 	}
 
-	//nolint:staticcheck
-	if err := spec.RewriteConstants(consts); err != nil {
-		return fmt.Errorf("rewriting constants: %w", err)
+	for varSpec, val := range consts {
+		if err := varSpec.Set(val); err != nil {
+			return fmt.Errorf("setting variable %s: %w", varSpec, err)
+		}
```

Comment:
`SpecSetVars` should be used when the caller doesn't have a struct generated with bpf2go (in the ebpf operator for image based gadgets for instance). It's not possible to use that helper since it relies on the const name, instead the logic here relies on the ebpf.VariableSpec fields on the collection spec.

To extend a bit more on this, I tried to use the new `VariableSpec` introduced in bpf2go, but I can revert it back to only relying on the const name too.